jparest/src/test/java/uk/gov/homeoffice/digital/sas/jparest/exceptions/ApiResponseExceptionHandlerTest.java

```diff
@@ -69,6 +70,23 @@ void handleUnexpectedQueryResultException_internalServerErrorWithErrorDataIsRetu
         assertResponseData(response, exception.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
     }
 
+    @Test
+    void handleResourceConstraintViolationException_badRequestWithErrorDataIsReturned() throws JSONException {
+        var error = new JSONObject();
+        error.put("field", "foo");
+        error.put("message", "bar");
+        error.put("data", null);
+        var errorResponse = new JSONObject[]{error};
+
+        var apiResponseExceptionHandler = new ApiResponseExceptionHandler();
+        var exception = new ResourceConstraintViolationException(errorResponse);
+        var response = apiResponseExceptionHandler.handleResourceConstraintViolationException(exception);
+
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+        assertThat(response.getBody()).isNotNull();
+        assertThat(response.getBody()).isEqualTo(errorResponse);
```

Comment:
nit: this can be chained (can apply to ln 93 too)
```suggestion
        assertThat(response.getBody()).isNotNull().isEqualTo(errorResponse);
```

Reply:
Ah yeah you're right, just this one is fine.